### ISSUE TITLE
Populating Custom Snippets

### ISSUE THREAD

**Issue by shshaw** [emmetio/codemirror6-plugin#13]
How do you add custom snippets to Emmet in CodeMirror 6?

In the past, we would pass through an `emmet` property on the CodeMirror config that would contain all the snippets. Obviously we can't do that directly in CodeMirror 6. Would it be a Facet or something?

**Comment by sergeche** [comment]
Just released v0.2.0 where you can pass custom snippets as well as change low-level Emmet behaviour. You have 2 options here: either pass config as argument of `abbreviationTracker` or use `emmetConfig` facet:

```js
new EditorView({
    doc: text,
    extensions: [
        // Pass options to tracker
        abbreviationTracker({
            config: {
                markup: {
                    snippets: {
                        'foo': 'ul.foo>li.bar+li.baz'
                    }
                }
            }
        }),
        // ...or use facet
        emmetConfig.of({
            config: {
                markup: {
                    snippets: {
                        'foo': 'ul.foo>li.bar+li.baz'
                    }
                }
            }
        }),
    ]
});
```

Note that options you pass is an object that contains some high-level _plugin_ options. This object may include `config` which is _Emmet core_ options. In this config, you specify a _per-syntax_ options for Emmet. E.g. you can specify own snippets and options for different syntaxes like `html`, `pug`, `css`, `scss` and so on. Also, you can common options for syntaxes of given type: `markup` (all markup syntaxes like HTML, XML, Pug, Slim etc.) or `stylesheet` (all stylesheet syntaxes like CSS, SCSS, etc).

```js
emmetConfig.of({
    config: {
        // A config for all markup languages like HTML, XML etc.
        markup: {
            snippets: {
                'foo': 'ul.foo>li.bar+li.baz'
            }
        },
        // A config for all HTML syntax (won’t apply to XML, for example)
        html: {
            snippets: {
                'foo2': 'ul.foo2>li.bar2+li.baz2'
            }
        }
    }
})
```

**Comment by shshaw** [comment]
Ah, that's helpful. Thank you. I'll give it a try in v0.2.0
